docs/en/03_Upgrading/01_Upgrading_project.md

```diff
@@ -338,36 +338,62 @@ You'll likely have some conflicts to resolve, whether you've updated your depend
 
 Running a `composer update` will tell you which modules are conflicted and suggested alternative combinations of modules that might work.
 
-The most typical reason for a conflict is that the maintainer of a module hasn't released a version compatible with SilverStripe 4.
+The most typical reason for a conflict is that the maintainer of a module has not released a version compatible with SilverStripe 4.
 
 If the maintainer of the module is in the process of upgrading to SilverStripe 4, a development version of the module might be available. In some cases, it can be worthwhile to look up the repository of the module or to reach out to the maintainer.
 
 <div class="info" markdown="1">
-If you're going to install development version of third party modules, you should consider adding the following entries to your `composer.json` file.
+If you're going to install a development version of third party modules, you should consider adding the following entries to your `composer.json` file:
 
 ```json
 {
   // ...
   "minimum-stability": "dev",
-  "prefer-stable": true,
+  "prefer-stable": true
   // ...
 } 
 ```
 </div>
 
-To resolve a conflict you can either:
-* fork the affected module and upgrade it yourself. Don't forget to send a pull request to the original module!
-* Integrate the affected module into your project's codebase
-* Remove the module from your project, if it is not essential
+If no development release is available for SilverStripe 4, you can upgrade the module manually or remove the module from your project.
 
-To integrate a third party module in your project, remove it from your `composer.json` file and from your `.gitignore` file. Then track the module's codebase in your project source control. You'll need to upgrade the module's code to be compatible with SilverStripe 4. 
+#### Upgrading the module manually
+
+To upgrade an incompatible module yourself, follow these steps:
+
+1. Integrate the affected module into your project's codebase
```

Comment:
This gives the impression that point 1 and point 2 have to be performed together. They are mutually exclusive.